inc/class-core-sitemaps-provider.php

```diff
@@ -245,12 +256,52 @@ public function get_sitemap_url( $name, $page ) {
 	/**
 	 * Get the last modified date for a sitemap page.
 	 *
+	 * This will be overridden in provider subclasses.
+	 *
 	 * @param string $name The name of the sitemap.
 	 * @param int    $page The page of the sitemap being returned.
 	 * @return string The GMT date of the most recently changed date.
 	 */
 	public function get_sitemap_lastmod( $name, $page ) {
-		return '0000-00-00 00:00Z GMT';
+		$type = implode( '_', array_filter( array( $this->slug, $name, (string) $page ) ) );
+
+		// Check for an option.
+		$lastmod = get_option( "core_sitemaps_lasmod_$type", '' );
+
+		// If blank, schedule a job.
+		if ( empty( $lastmod ) && ! wp_doing_cron() ) {
+			wp_schedule_single_event( time() + 500, 'core_sitemaps_calculate_lastmod', array( $this->slug, $name, $page ) );
+		}
```

Comment:
This can result in a blank lastmod value rendering into the sitemap. Consider setting a fallback value to `parent::get_sitemap_lastmod()` after scheduling a job, so that there is always a valid timestamp.

Reply:
This is the intended behavior. Since `lastmod` values for sitemap pages is an expensive operation, we want to show no value when we don't have a previously calculated value, so we aren't blocking rendering of the page. When this is the case, we calculate this value asynchronously so it's available the next time the index is displayed.

Reply:
Ah I missed that `lastmod` is an optional element, so it can be omitted safely. All good.